lib/rules/no-restricted-invocations.js

```diff
@@ -76,18 +85,19 @@ module.exports = class NoRestrictedInvocations extends Rule {
     };
   }
 
-  _checkNode(node, name, message) {
+  _getComponentOrHelperName(node) {
     if (this.isLocal(node)) {
-      return;
+      return undefined;
     }
 
     if (node.type === 'ElementNode') {
-      if (dasherize(node.tag) === name || node.tag === name) {
-        this._logNode(node, `<${node.tag} />`, message);
-      }
+      // Convert from angle-bracket naming to kebab-case.
+      return dasherize(node.tag);
     } else {
-      if (node.path.original === name || checkForComponentHelper(node, name)) {
-        this._logNode(node, `{{${name}}}`, message);
+      if (node.path.original === 'component' && node.params[0]) {
+        return node.params[0].original;
```

Comment:
Not really relevant to these changes, but it would be easy to also check for `(component (component ('disallowed-component')))` which could be used as a (terrible) workaround for this lint rule by making this a `while` loop.